ansible/roles/base/tasks/main.yml

```diff
@@ -10,6 +10,26 @@
 - name: Set DHCP config
   template: src=etc/dhcp/dhclient.conf.j2 dest=/etc/dhcp/dhclient.conf
 
+# Node
+- name: Register NodeJS
+  shell: 'curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -'
+  changed_when: false
+
+# Yarn
+- name: Install Yarn APT key
+  become: true
+  apt_key:
+    url: 'https://dl.yarnpkg.com/debian/pubkey.gpg'
+    state: present
+
+- name: Add Yarn to the source lists
+  become: true
+  lineinfile:
```

Comment:
There is a module to simplify adding repos. It's considered best practice to use the modules where available, unless there is a specific reason not to do so.